### ISSUE TITLE
Two options to set the fan speed that needed to be set both

### ISSUE THREAD

**Issue by inovatorius** [supermerill/SuperSlicer#4146]
### What happened?

So, default values for the fans are 100%
![image](https://github.com/supermerill/SuperSlicer/assets/122360472/ef075d6f-47ff-4656-8194-e323a1d83880)
![image](https://github.com/supermerill/SuperSlicer/assets/122360472/126da176-1785-40a2-bf7e-d9a2fcc0b709)

If you set "Default fan speed" to 50% nothing happens nor if you set "Max fan speed" to 50%. You have to set them both to 50% in order to fan speed to take the effect event though all the other _speeds_ are set to -1 which should just use the "Default fan speed" (at least that is what description tooltip says)
![image](https://github.com/supermerill/SuperSlicer/assets/122360472/28680f62-32d6-4568-a9c0-1d868db991af)
![image](https://github.com/supermerill/SuperSlicer/assets/122360472/e3668239-5925-4d54-b9ce-c42a1875971e)

### Project file & How to reproduce

[Shape-Cylinder.zip](https://github.com/supermerill/SuperSlicer/files/14366581/Shape-Cylinder.zip)

**Comment by jbeima** [comment]
Unless I am missing something here. I believe this is working correctly. Let me share my mindset on this one.

First the "Max Fan Speed" that you are point to applies to layers that fall under the "Short Layer Time". Since I would believe all of those layers are printing in under 100 second. They would all be classed as short layer times, thus the fan would run at the "short layer time" - "Max Fan Speed".

You can test this by changing that short layer time - time down to say 10 seconds from 100, just to see how it slices, or even 0 to disable it. Just for the sake of seeing how it effects things.

This one can trip people up. Since you also have the "Very Short Layer Time" overrides just below it as well.

:)

**Comment by supermerill** [comment]
Current behavior:
 - default speed when possible
 - if below the "short layer time", then it will increase up to the max (here it can't as it's already higher)
 - if below the "very short layer time", it will go to the maximum speed between the current one and the max.
 
 Why?
 because for some feature like bridge, you may want to go higher than any reasonable fan speed, as the layer adhesion isn't important for bridge, as there isn't anything under it.

**Comment by inovatorius** [comment]
There is a plenty setting we can adjust and it is a good thing, however looking at the Prusa Slicer UI for the Cooling options it is more obvious what you can change and enable or disable:
<img width="779" alt="image" src="https://github.com/supermerill/SuperSlicer/assets/122360472/654659c5-7468-4971-aebd-b9c044e69a03">

That "Enable auto cooling" checkbox is really doing what I need without me needing to read all that descriptions in the tooltips that after reading adding a bit more confuss.

Also, it is unclear what options can be disabled for example, this one:
<img width="654" alt="image" src="https://github.com/supermerill/SuperSlicer/assets/122360472/776dbd20-d084-494a-a51f-86cb2531f8c4">

Near the "approximate seconds" line there should be "(zero to disable)" just like in other options:
<img width="294" alt="image" src="https://github.com/supermerill/SuperSlicer/assets/122360472/b1f0f5dd-10dc-4d81-8fa7-965a944a8b34">

And one thing is these options like 0 or -1. This is getting out of hand. Can we just settle with 0 is disable because sometimes -1 is to disable things or sometimes it is 1.
<img width="766" alt="image" src="https://github.com/supermerill/SuperSlicer/assets/122360472/dd1d5b24-e456-4651-9c3e-4d8c03c0e9ac">

I cannot recommend anything at this point, because there is just no consistency and I need to take some time to deep dive further into SuperSlicer to actually provide something insightful. I do understand, that all of this was probably discussed already and there is a clear logic behind.

**Comment by inovatorius** [comment]
Other thing to mention, M106 and M107 are being spammed all over the place:
<img width="1023" alt="Screenshot 2024-02-27 at 03 00 40" src="https://github.com/supermerill/SuperSlicer/assets/122360472/6709133d-f2ce-439b-bf85-6383ec72f89f">
<img width="1019" alt="Screenshot 2024-02-27 at 03 02 21" src="https://github.com/supermerill/SuperSlicer/assets/122360472/65f89eb9-a5ad-4415-965d-db8b41bddfc8">

**Comment by supermerill** [comment]
> And one thing is these options like 0 or -1. This is getting out of hand. 

yes, It's not ideal. I may have to redo the gui & settings to add checkboxes for a proper "disable" instead of these values.

> In this example, if we remove M107 we save around 2000 lines of gcode. (31K vs 33K lines)

Yes, known problem. Currently, the fan_mover should post-process the output and remove any unnecessary fan command.
I didn't always activated yet because even if I added many test & many debug I'm not 100% confident it's empty of bugs, but i'll do in the 2.7
